engine/src/main/java/io/zeebe/engine/processor/AsyncSnapshotDirector.java

```diff
@@ -234,14 +229,17 @@ private void onCommitCheck() {
                         (processedPosition, ex2) -> {
                           if (ex2 == null) {
                             enforceSnapshotCreation(writtenPosition, processedPosition);
+                            metrics.close();
                             actor.close();
```

Comment:
We probably want to close the actor and metrics anyway, even if the snapshot was not successful ? What do you think